### PR TITLE
drop six requirement

### PR DESCRIPTION
**Pull request by branchvincent**
I noticed `six` is still required/used even though only 3.6+ is supported

* First commit is all from [pyupgrade](https://github.com/asottile/pyupgrade): `pipx run pyupgrade --py36-plus $(git ls-files '*.py')`
* Second commit is manual fixes, replacing the remaining uses of `six` and updating requirements

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -2,10 +2,9 @@ docker
 PyYAML
```
Comment (Gallaecio) [inline requirements.in:2]: We should probably remove `requirements.in` and `requirements.txt` altogether.

File: shub/image/deploy.py [hunk 2]
```diff
@@ -133,10 +133,10 @@ class _LoggedDeployProgress(utils.BaseProgress):
 
     def show(self):
         click.echo("Deploy results:")
-        super(_LoggedDeployProgress, self).show()
+        super().show()
 
     def handle_event(self, event):
-        click.echo("{}".format(event))
+        click.echo(f"{event}")
```
Comment (Gallaecio) [inline shub/image/deploy.py:139]: If:

```suggestion
        click.echo(event)
```

Does not work, then maybe:

```suggestion
        click.echo(str(event))
```

File: shub/image/test.py [hunk 3]
```diff
@@ -1,13 +1,10 @@
...
-
 import click
 
 from shub import exceptions as shub_exceptions
 from shub.config import load_shub_config, list_targets_callback
 from shub.image import utils
 
-if sys.version_info >= (3, 0):
-    long = int
+long = int
```
Comment (Gallaecio) [inline shub/image/test.py:7]: We can remove this line altogether and replace `isinstance(size, (int, long))` with `isinstance(size, int)` later on.

File: shub/image/test.py [hunk 4]
```diff
@@ -84,7 +81,7 @@ def _check_image_size(image_name, docker_client):
             raise shub_exceptions.CustomImageTooLargeException(
                 IMAGE_TOO_LARGE_WARNING)
     except NotFound as exc:
-        utils.debug_log("{}".format(exc))
+        utils.debug_log(f"{exc}")
```
Comment (Gallaecio) [inline shub/image/test.py:84]: Same as with the earlier echo, `exc` may work, else `str(ext)`.

File: tests/test_utils.py [hunk 5]
```diff
@@ -1,9 +1,3 @@
-#!/usr/bin/env python
```
Comment (Gallaecio) [inline tests/test_utils.py:1]: I noticed this on other test files. We could probably remove it as well from all of them. But I’m fine with keeping that out of the scope of these changes.

**Review by Gallaecio — APPROVED** [review]
:heart:

File: shub/image/deploy.py (hunk 2 above)
Comment (branchvincent) [reply shub/image/deploy.py:139]: `click.echo` casts args to str so first one works

File: requirements.in (hunk 1 above)
Comment (branchvincent) [reply requirements.in:2]: dropped

File: tests/test_utils.py (hunk 5 above)
Comment (branchvincent) [reply tests/test_utils.py:1]: removed remaining shebangs

**Comment by branchvincent** [comment]
btw, any interest in raising the minimum python to 3.8 (since older versions are EOL)?

**Comment by wRAR** [comment]
Yeah, I don't think it's useful to keep support for pre-3.8.

File: tests/test_utils.py (hunk 5 above)
Comment (wRAR) [reply tests/test_utils.py:1]: Can you please also remove `if __name__ == '__main__':     unittest.main()` blocks from those test files?

File: shub/image/run/wrapper.py [hunk 6]
```diff
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (branchvincent) [reply shub/image/run/wrapper.py]: oops, restored

**Review by wRAR — APPROVED** [review]

**Comment by branchvincent** [comment]
Rebased to resolve conflicts after #434

**Review by wRAR — APPROVED** [review]

**Review by Gallaecio — APPROVED** [review]

**Pull request merged**